Source/TestDafny/MultiBackendTest.cs

```diff
@@ -153,13 +200,80 @@ public class MultiBackendTest {
 
     if (success) {
       output.WriteLine(
-        $"All executions were successful and matched the expected output (or reported errors for known unsupported features)!");
+        "All executions were successful and matched the expected output (or reported errors for known unsupported features)!");
       return 0;
     } else {
       return -1;
     }
   }
 
+  public int ForEachResolver(ForEachResolverOptions options) {
+    var parseResult = CommandRegistry.Create(TextWriter.Null, TextWriter.Null, TextReader.Null,
+      new string[] { "verify", options.TestFile! }.Concat(options.OtherArgs).ToArray());
+    var dafnyOptions = ((ParseArgumentSuccess)parseResult).DafnyOptions;
+
+    // We also use --(r|b)print to catch bugs with valid but unprintable programs.
+    string fileName = Path.GetFileName(options.TestFile!);
+    var testDir = Path.GetDirectoryName(options.TestFile!);
+    var tmpDPrint = Path.Join(testDir, "Output", $"{fileName}.dprint");
+    var tmpRPrint = Path.Join(testDir, "Output", $"{fileName}.rprint");
+    var tmpPrint = Path.Join(testDir, "Output", $"{fileName}.print");
+
+    var dafnyArgs = new List<string>() {
+      $"verify",
+      options.TestFile!,
+      $"--print:{tmpDPrint}",
+      $"--rprint:{tmpRPrint}",
+      $"--bprint:{tmpPrint}"
+    }.Concat(options.OtherArgs.Where(OptionAppliesToVerifyCommand)).ToArray();
+
+    output.WriteLine("Verifying...");
+
+    var resolutionOptions = new List<ResolutionSetting>() {
+      new ResolutionSetting(new string[] { }, new string[] { ".expect" },
+        options.ExpectExitCode != null ? (int)options.ExpectExitCode : 0),
+      new ResolutionSetting(new string[] { "--type-system-refresh" }, new string[] { ".refresh.expect", ".expect" },
+        options.RefreshExitCode != null ? (int)options.RefreshExitCode : options.ExpectExitCode != null ? (int)options.ExpectExitCode : 0)
```

Comment:
nit: can be simplified to
```suggestion
      new ResolutionSetting(new string[] { }, new string[] { ".expect" },
        (int)(options.ExpectExitCode ?? 0)),
      new ResolutionSetting(new string[] { "--type-system-refresh" }, new string[] { ".refresh.expect", ".expect" },
        (int)(options.RefreshExitCode ?? options.ExpectExitCode ?? 0))
```